starfive/visionfive/v2/default.nix

```diff
@@ -1,15 +1,10 @@
 { config, lib, pkgs, ... }: {
   boot = {
-    # Force no ZFS (from nixos/modules/profiles/base.nix) until updated to kernel 6.0
-    supportedFilesystems =
-      lib.mkForce [ "btrfs" "reiserfs" "vfat" "f2fs" "xfs" "ntfs" "cifs" ];
     consoleLogLevel = lib.mkDefault 7;
-    kernelPackages = lib.mkDefault (pkgs.callPackage ./linux-6.6.nix {
-      inherit (config.boot) kernelPatches;
-    });
+    # Require at least Linux >= 6.9
+    kernelPackages = lib.mkDefault pkgs.linuxPackages_latest;
```

Comment:
Could you rebase on top of the latest nixos-hardware and add the remaining kernel-patch from https://github.com/NixOS/nixos-hardware/pull/958/files?

Reply:
This uses the older v2 patch. Should we get a newer kernel than 6.9.2 we may have to apply v3 of that patch: https://patchwork.kernel.org/project/linux-riscv/patch/20240605071701.13737-1-hal.feng@starfivetech.com/

Reply:
do you think the patch will enable further hardware on 6.9?

Reply:
> do you think the patch will enable further hardware on 6.9?

Hal's defconfig patch v3 notes in changes from v2 "Add CONFIG_MOTORCOMM_PHY=m, which supports the ethernet phy chips on the board."

Otherwise you get a networking interface that may or may not be functional, depending on whether U-Boot initialized it with its own motorcomm PHY driver code and/or if the device was powered off or not during the previous reboot.

Reply:
@eshattow Does this fix the usb issue? And would you know how to apply the PCIe patches https://patchwork.kernel.org/project/linux-riscv/patch/20240328091835.14797-23-minda.chen@starfivetech.com/ ? Here i'm a bit confused: The statuspage https://rvspace.org/en/project/JH7110_Upstream_Plan lists PCIe as "Accepted", but the patchwork status has status "new", which reads like unreviewed and unmerged.

Reply:
@malteneuss I'm not very familiar with Linux Kernel development; any patches that Linux Kernel subsystem maintainers have accepted are in the form of a git branch that Linus will pull into mainline at the appropriate time. This is the situation with the PCIe patches, they are accepted, and queued up waiting for Linus to pull them during the next merge window (on whatever schedule of time that may be).

Please re-read [my above comment](https://github.com/NixOS/nixos-hardware/pull/964#issuecomment-2153332424) following and reading all described links. You will find your answers.

Reply:
As a follow-up confirming all relevant Linux kernel config and patch sets (including devicetree) for PCIe functionality on the common JH7110 boards (VisionFive2, Mars*, Star64) are queued up. This should all land in 6.11rc1. Similarly as an aside, the first rc1 after the recent U-Boot v2024.07 release will have the EFI-related plumbing to load devicetree fdt from files named by EFI variables and/or U-Boot environment variables.

Reply:
> As a follow-up confirming all relevant Linux kernel config and patch sets (including devicetree) for PCIe functionality on the common JH7110 boards (VisionFive2, Mars*, Star64) are queued up. This should all land in 6.11rc1. Similarly as an aside, the first rc1 after the recent U-Boot v2024.07 release will have the EFI-related plumbing to load devicetree fdt from files named by EFI variables and/or U-Boot environment variables.

Merged in mainline Linux as https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=3f386cb8ee9f04ff4be164ca7a1d0ef3f81f7374 (pre-6.11rc1).

JH7110 Cadence USB is confirmed functional since Linux v6.5 but depends on support at U-Boot phase that is only recently proposed and under review; This differs from the Starfive board support package and derivatives which apply workarounds within Linux kernel driver code. It's not clear to me how this co-dependency on U-Boot will apply to Linux devicetree changes that are needed to be applied, but otherwise there's nothing that needs to be done in Linux for USB support the drivers have been there awhile and it is just this JH7110 CPU and board-specific errata that needs to be set up _somewhere_.

Reply:
I'm going to give Kernel 6.11-rc1 a try as soon it will be released but not sure which U-Boot or device tree patches are required for USB to work

Reply:
@onny I kind of figure that the U-Boot stuff is out-of-scope for this issue. Welcome to continue discussion (and I have examples I would like Star64 owners to do testing with) at the [Star64 Matrix chat](https://matrix.to/#/#star64:matrix.org)